cql3/statements/describe_statement.cc

```diff
@@ -142,6 +144,82 @@ future<std::vector<description>> generate_descriptions(
     co_return descs;
 }
 
+template <std::ranges::range Range>
+    requires std::is_base_of_v<keyspace_element, std::ranges::range_value_t<Range>>
+future<std::vector<description>> generate_descriptions(replica::database& db, const Range& elements,
+    std::optional<bool> with_internals = std::nullopt, bool sort_by_name = true)
+{
+    std::vector<description> descs{};
+
+    // In general, a range is allowed to be modified during the call to `std::ranges::size()`.
+    // However, we rely on the sanity of the passed types and the fact that we take the range
+    // as a const-reference.
+    descs.reserve(std::ranges::size(elements));
+
+    for (auto& e: elements) {
+        auto desc = (with_internals.has_value())
+            ? description(db, e, *with_internals)
+            : description(db, e);
+        descs.push_back(std::move(desc));
+
+        co_await coroutine::maybe_yield();
+    }
+
+    if (sort_by_name) {
+        std::ranges::sort(descs, [] (const auto& lhs, const auto& rhs) {
+            return lhs._name < rhs._name;
+        });
+    }
+
+    co_return descs;
+}
+
+future<std::vector<description>> generate_descriptions_with_generator(const data_dictionary::keyspace_element_generator& generator,
+        const replica::database& db, const bool with_internals, const bool sort_by_name = true)
+{
+    std::vector<description> result{};
+
+    const sstring keyspace_name = generator.keyspace_name();
+    const sstring element_type = generator.element_type(db);
+
+    auto described_elements = co_await generator.describe_elements(db, with_internals);
+    result.reserve(described_elements.size());
+
+    for (auto& [element_name, create_statement] : described_elements) {
+        result.emplace_back(keyspace_name, element_type, std::move(element_name), std::move(create_statement));
+        co_await coroutine::maybe_yield();
+    }
+
+    if (sort_by_name) {
+        std::ranges::sort(result, [] (const auto& lhs, const auto& rhs) {
+            return lhs._name < rhs._name;
+        });
+    }
+
+    co_return result;
+}
+
+template <std::ranges::range Range>
+    requires std::is_base_of_v<data_dictionary::keyspace_element_generator, std::ranges::range_value_t<Range>>
+future<std::vector<description>> generate_descriptions_with_multiple_generators(const Range& generators,
```

Comment:
Yes, unfortunately (but chances are I just don't see a way to avoid a template not having to create an artificial `std::vector<std::reference_wrapper<base_type>>`)